pkg/apis/flowcontrol/types.go

```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	APIGroupAll    = "*"
+	ResourceAll    = "*"
+	NonResourceAll = "*"
+
+	VerbAll = "*"
+
+	GroupKind          = "Group"
+	ServiceAccountKind = "ServiceAccount"
+	UserKind           = "User"
+)
+
+// These are valid condition statuses. "ConditionTrue" means a resource is in the condition;
+// "ConditionFalse" means a resource is not in the condition; "ConditionUnknown" means kubernetes
+// can't decide if a resource is in the condition or not. In the future, we could add other
+// intermediate conditions, e.g. ConditionDegraded.
+const (
+	FlowSchemaConditionTrue    FlowSchemaConditionStatus = "True"
+	FlowSchemaConditionFalse   FlowSchemaConditionStatus = "False"
+	FlowSchemaConditionUnknown FlowSchemaConditionStatus = "Unknown"
+
+	PriorityLevelConfigurationConditionTrue    PriorityLevelConfigurationConditionStatus = "True"
+	PriorityLevelConfigurationConditionFalse   PriorityLevelConfigurationConditionStatus = "False"
+	PriorityLevelConfigurationConditionUnknown PriorityLevelConfigurationConditionStatus = "Unknown"
+)
+
+const (
+	FlowDistinguisherMethodByUserName  FlowDistinguisherMethod = "ByUserName"
+	FlowDistinguisherMethodByUserGoup  FlowDistinguisherMethod = "ByUserGroup"
```

Comment:
What is this method?  An authorized user has a general number of groups.

Reply:
removed